### DISCUSSION TITLE
4.x, 分组柱状图，分组内的排序逻辑是怎样的？

### DISCUSSION THREAD

**Discussion by tobemaster56** [antvis/G2#6097] [Q&A]
demo：
[codesandbox](https://codesandbox.io/p/sandbox/modest-hugle-mc2qnt?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clt3srgz600063b6evb00974g%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clt3srgz500023b6ewoemfswx%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clt3srgz500033b6eh5xtro75%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clt3srgz500053b6ecg1er6wi%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clt3srgz500023b6ewoemfswx%2522%253A%257B%2522id%2522%253A%2522clt3srgz500023b6ewoemfswx%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clt3su4v300503b6e9ceypqkn%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A3%252C%2522startColumn%2522%253A18%252C%2522endLineNumber%2522%253A3%252C%2522endColumn%2522%253A24%257D%255D%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clt3su4v300503b6e9ceypqkn%2522%257D%252C%2522clt3srgz500053b6ecg1er6wi%2522%253A%257B%2522id%2522%253A%2522clt3srgz500053b6ecg1er6wi%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clt3srgz500043b6e0l9t8phw%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clt3srgz500043b6e0l9t8phw%2522%257D%252C%2522clt3srgz500033b6eh5xtro75%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clt3srgz500033b6eh5xtro75%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D
)

数据的顺序
b 对应的分组顺序为 ["52", "0", "-75"]
<img width="549" alt="image" src="https://github.com/antvis/G2/assets/8998761/8d3cfed1-27d6-48e8-bbcc-b92d92cc77a5">

但是，实际渲染的顺序是  ["0",  "52", "-75"]
<img width="522" alt="image" src="https://github.com/antvis/G2/assets/8998761/4172f1f1-0cb8-4c2a-ba3c-5b5c3ca362f5">

**Comment by hustcc** [comment]
按照 color 字段咋数据中出现顺序来的。

**Reply by tobemaster56** [reply]
请问，有办法可以按照数据实际的顺序来吗？
